tests/test_github.py

```diff
@@ -105,6 +118,96 @@ def test_github_retarget_pr(self) -> None:
             strip_indentation=False
         )
 
+    @mock.patch('git_machete.cli.exit_script', mock_exit_script)
+    @mock.patch('git_machete.github.GITHUB_REMOTE_PATTERNS', FAKE_GITHUB_REMOTE_PATTERNS)
+    @mock.patch('git_machete.utils.run_cmd', mock_run_cmd)  # to hide git outputs in tests
+    @mock.patch('urllib.request.Request', git_api_state_for_test_retarget_pr.new_request())
+    @mock.patch('urllib.request.urlopen', MockContextManager)
+    def test_github_retarget_pr_multiple_non_origin_remotes(self) -> None:
+        branchs_first_commit_msg = "First commit on branch."
+        branchs_second_commit_msg = "Second commit on branch."
+
+        origin_1_remote_path = mkdtemp()
+        origin_2_remote_path = mkdtemp()
+        origin_3_remote_path = mkdtemp()
+        self.repo_sandbox.new_repo(origin_1_remote_path, "--bare")
+        self.repo_sandbox.new_repo(origin_2_remote_path, "--bare")
+        self.repo_sandbox.new_repo(origin_3_remote_path, "--bare")
+
+        os.chdir(self.repo_sandbox.local_path)
+
+        # branch feature present in all of the remotes
+        (
+            self.repo_sandbox.remove_remote(remote='origin')
+                .new_branch("root")
+                .add_remote('origin_1', origin_1_remote_path)
+                .add_remote('origin_2', origin_2_remote_path)
+                .add_remote('origin_3', origin_3_remote_path)
+                .commit("First commit on root.")
+                .push(remote='origin_1')
+                .push(remote='origin_2')
+                .push(remote='origin_3')
+                .new_branch("branch-1")
+                .commit(branchs_first_commit_msg)
+                .commit(branchs_second_commit_msg)
+                .push(remote='origin_1')
+                .push(remote='origin_2')
+                .push(remote='origin_3')
+                .new_branch('feature')
+                .commit('introduce feature')
+                .push(remote='origin_1')
+                .push(remote='origin_2')
+                .push(remote='origin_3')
+        )
+
+        launch_command("discover", "-y")
+        expected_error_message = "Multiple non-origin remotes correspond to GitHub in this repository: origin_1, origin_2, origin_3 -> aborting. \nYou can also select the repository by providing 3 git config keys: `machete.github.{remote,organization,repository}`\n"
+        with pytest.raises(MacheteException) as e:
+            launch_command("github", "retarget-pr")
+        if e:
+            assert e.value.args[0] == expected_error_message, 'Verify that expected error message has appeared when given pull request to create is already created.'
+
+        # branch feature_2 present in none of the remotes
+        (
+            self.repo_sandbox.check_out('feature')
+                .new_branch('feature_2')
+                .commit('introduce feature 2')
+        )
+
+        launch_command("discover", "-y")
```

Comment:
Hmmm is that `discover` after every single command needed here? does it change anything in the contents of `.git/machete`? I'd rather remove it if possible to unclutter/clarify the test